### DISCUSSION TITLE
v7.5.5 includes broken justinrainbow/json-schema?

### DISCUSSION THREAD

**Discussion by wilfriedwolf** [deployphp/deployer#3969] [Bugs] [answered]
### Deployer Version

v7.5.6

### Target OS

Ubuntu 24.04

### Which PHP version are you using?

PHP 8.1

### Content of deploy.php or deploy.yaml

tasks:
  echo:
    - run: 'echo test'

hosts:
  localhost:
    local: true

### Steps to reproduce

$ dep echo localhost
 ErrorException  in BaseConstraint.php on line 46:

  Optional parameter $path declared before required parameter $message is implicitly treated as a required parameter

No problem with v7.5.5!

**Comment by dirkj** [comment]
I observed the same behavior since yesterday. Following additions:

- Running MacOS with PHP 8.3 (tried 8.2 and 8.4 as well)
- Problem occurs only in project directory, not within other directory. "dep --version" works perfectly everywhere else, but not in my project directory. There I get the exception on BaseConstraint.php as well (Optional parameter $path declared before required parameter $message is implicitly treated as a required parameter)
- Tried to up/downgrade several packages in my composer setup in my project (used composer.lock/composer.json version, that worked a day before) -- without success

Any hint or idea what the cause is?

**Comment by TheRealJanJanssens** [comment]
We are experiencing the same issue. Downgrading to v7.5.5 temporarily resolved the issue for us.
Seems like json-schema package included in the source code is causing the problem.

**Comment by TheRealJanJanssens** [comment]
Here is your culprit:
https://github.com/deployphp/deployer/compare/v7.5.6...7.x?diff=split&w=

The change done on BaseConstraint.php causes the error since optional parameters before required parameters has been deprecated since PHP 8.0.

**Comment by Defcon0** [comment]
Same here

**Comment by antonmedv** [comment]
Ok, I think I fixed all errors. Use this to find this regex:
```
grep -irnE --include \*.php 'function.+?=.+?\$[^=]+,'
```

**Comment by antonmedv** [comment, marked as answer]
Should be fixed in v7.5.7.
